python/demo/poisson/demo_poisson.py

```diff
@@ -1,204 +1,236 @@
+# is the linear form. It is assumed that all functions in $V$
+# satisfy the Dirichlet boundary conditions ($u = 0 \ {\rm on} \
+# \Gamma_{D}$).
 #
 # In this demo, we shall consider the following definitions of the input
 # functions, the domain, and the boundaries:
 #
-# * :math:`\Omega = [0,1] \times [0,1]` (a unit square)
-# * :math:`\Gamma_{D} = \{(0, y) \cup (1, y) \subset \partial \Omega\}`
+# - $\Omega = [0,2] \times [0,1]$ (a rectangle)
+# - $\Gamma_{D} = \{(0, y) \cup (1, y) \subset \partial \Omega\}$
 #   (Dirichlet boundary)
-# * :math:`\Gamma_{N} = \{(x, 0) \cup (x, 1) \subset \partial \Omega\}`
+# - $\Gamma_{N} = \{(x, 0) \cup (x, 1) \subset \partial \Omega\}$
 #   (Neumann boundary)
-# * :math:`g = \sin(5x)` (normal derivative)
-# * :math:`f = 10\exp(-((x - 0.5)^2 + (y - 0.5)^2) / 0.02)` (source
+# - $g = \sin(5x)$ (normal derivative)
+# - $f = 10\exp(-((x - 0.5)^2 + (y - 0.5)^2) / 0.02)$ (source
 #   term)
 #
-#
-# Implementation
-# --------------
+# ## Implementation
 #
 # This description goes through the implementation (in
-# :download:`demo_poisson.py`) of a solver for the above described
+# {download}`demo_poisson.py`) of a solver for the above described
 # Poisson equation step-by-step.
 #
-# First, the :py:mod:`dolfinx` module is imported: ::
+# First, the {py:mod}`dolfinx` module is imported:
 
+# +
 import numpy as np
 
 import ufl
-from dolfinx import fem, plot
-from dolfinx.fem import FunctionSpace, dirichletbc, locate_dofs_topological
-from dolfinx.io import XDMFFile
-from dolfinx.mesh import CellType, create_rectangle, locate_entities_boundary
-from ufl import ds, dx, grad, inner
+from dolfinx import fem, io, mesh, plot
+from ufl import ds, dx, exp, grad, inner, sin
 
 from mpi4py import MPI
 from petsc4py.PETSc import ScalarType
+# -
 
 # We begin by defining a mesh of the domain and a finite element
-# function space :math:`V` relative to this mesh. As the unit square is
-# a very standard domain, we can use a built-in mesh provided by the
-# class :py:class:`create_unit_square_mesh
-# <dolfinx.mesh.create_unit_square_mesh>`. In order to create a mesh
-# consisting of 32 x 32 squares with each square divided into two
-# triangles, we do as follows ::
-
-# Create mesh and define function space
-mesh = create_rectangle(MPI.COMM_WORLD, ((0.0, 0.0), (2.0, 1.0)), (32, 16), CellType.triangle)
-V = FunctionSpace(mesh, ("Lagrange", 1))
-
-# The second argument to :py:class:`FunctionSpace
-# <dolfinx.fem.FunctionSpace>` is the finite element family, while the
-# third argument specifies the polynomial degree. Thus, in this case,
-# our space ``V`` consists of first-order, continuous Lagrange finite
-# element functions (or in order words, continuous piecewise linear
-# polynomials).
+# function space $V$ relative to this mesh. We create a rectangular mesh
+# using built-in function provided by class
+# {py:class}`create_rectangle<dolfinx.mesh.create_rectangle>`. In order
```

Comment:
This should be a `{py:func}`?

Reply:
Yes there are lots of mistakes :) This conversion was done (almost) automatically. The purpose of this PR is to get the infrastructure in place, we can look at content on another PR.